### ISSUE TITLE
Fix S4830 FN: Local function statements are not supported

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4405]
### Description

S4830 rule does not support local function declarations

### Repro steps

```
static HttpWebRequest CreateRQ() // static local function
{
    return (HttpWebRequest)System.Net.HttpWebRequest.Create("http://localhost");
}

void InitAsArgument(RemoteCertificateValidationCallback callback)
{
    CreateRQ().ServerCertificateValidationCallback += callback; // FN, should be Non-compliant
}

InitAsArgument((sender, certificate, chain, SslPolicyErrors) => true);  //should be a secondary location
```

### Expected behavior

Issue with a corresponding secondary location should be raised here.

### Actual behavior

Issue is not being raised.

### Related information

* C#/VB.NET Plugins version 8.22
